test/test_binomial.cpp

```diff
@@ -716,6 +716,14 @@ void test_spots(RealType T)
 
    check_out_of_range<boost::math::binomial_distribution<RealType> >(1, 1); // (All) valid constructor parameter values.
 
+  {
+     using namespace boost::math::policies;
+     typedef policy<discrete_quantile<integer_round_outwards> > Policy;
+     binomial_distribution<RealType, Policy> dist(9079765771874083840, 0.561815);
+     // Accuracy is not too important here; the main purpose is to make
+     // sure it is not stuck.
+     BOOST_CHECK_CLOSE(quantile(dist, 0.0365346), 5101148604445670400, 1e12);
```

Comment:
On MSVC platforms you are getting an overflow here: https://github.com/boostorg/math/actions/runs/5721216709/job/15503232562#step:11:2377. `boost::math::concepts::real_concept` is a long double wrapped in a class to check conceptual requirements for support of user defined types.

Reply:
@mborland It's an unrelated bug in https://github.com/boostorg/math/blob/e682c1be3326a7b4ead43ff1c50149a4ce226e2e/include/boost/math/special_functions/beta.hpp#L489, more specifically `log(x * c / a) + b * log(y * c / b))` is less accurate on MSVC because `long double` is just `double` there.

Do you think it's acceptable to skip `real_concept` test on MSVC for this test case?

Reply:
I ran you branch locally on an M1 Mac (double = long double) and that works fine. I should have time this week to investigate why MSVC wants to be different.

Reply:
@mborland Note that in the newest version I skip the test for `real_concept` if `long double` is not precise enough.  I can reproduce the overflow on Linux if I force `real_concept_base_type` to be `double`.